### PR TITLE
deviceprofile controller update: support Create/Delete/Get between ed…

### PR DESCRIPTION
**Pull request by Peeknut**
Update deviceprofile controller and syncer:
1. deviceprofile created/deleted in kubernetes could sync in edgex-foundry.
2. deviceprofile create/deleted in edgex-foundry could sync in kubernetes.
Note: Can't support update opperation yet.

### CODE REVIEW

**Review by rambohe-ch — APPROVED** [review]
overall, LGTM

**Comment by rambohe-ch** [comment]
@Peeknut please fix the conflicting files.

File: clients/interface.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2021 The Kubernetes authors.
```
Comment (rambohe-ch) [inline clients/interface.go:2]: Kubernetes --> OpenYurt?

File: controllers/deviceprofile_syncer.go [hunk 2]
```diff
@@ -2,43 +2,51 @@ package controllers
 
 import (
 	"context"
+	"fmt"
```
Comment (rambohe-ch) [inline controllers/deviceprofile_syncer.go:5]: please add license header for all files.

File: controllers/util/util.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2021 The Kubernetes authors.
```
Comment (rambohe-ch) [inline controllers/util/util.go:2]: Please check all copyright, kubernetes should be modified to OpenYurt.

File: clients/interface.go (hunk 1 above)
Comment (Peeknut) [reply clients/interface.go]: done

File: controllers/deviceprofile_syncer.go (hunk 2 above)
Comment (Peeknut) [reply controllers/deviceprofile_syncer.go:21]: done

File: controllers/util/util.go (hunk 3 above)
Comment (Peeknut) [reply controllers/util/util.go]: done

**Pull request merged**